### PR TITLE
Documents bolt_types and fixes Mosin Nagant Sprite

### PR DESCRIPTION
**Pull request by Stonetear**
## About The Pull Request
For an unknown period of time, `/obj/item/gun/ballistic` used the wrong type of sprite for it's `bolt_type`, so anyone trying to make a new gun and using it as an example wouldn't be able to make it work.  Being an abstract item, it used `BOLT_TYPE_STANDARD` when it had a `BOLT_TYPE_LOCKING` icon state.

If you spawn it in and look at it, it looks like this.
image
But if you VV `bolt_bolt` type to `4` ( `BOLT_TYPE_LOCKING`), it looks correct.
image
Same with the Mosin, which now has a visible bolt, so you can tell if it's open or not now!
bolt

So basically:
- Changes the Mosin to use the correct bolt type, `BOLT_TYPE_LOCKING`.
- Makes the abstract `/obj/item/gun/ballistic` use the correct type of sprite.  It comes in designer _Debug Purple checkerboard_.
image
- Adds `bolt_types_explained.md`.  As the name implies, this explains in detail what all the bolt types do, and what you should name the iconstates for them when you're trying to make a new gun.
- Various comment changes in `ballistic.dm`, and the `combat.dm` defines.
- De-souls the `boolets` variable name to `bullets` in the ammo counter proc
- Adds a new parameter `handle_modifications = TRUE` to `sawoff()`  When set to false, it allows you to still use the sawing logic, but bypasses all the snowflake code for variable changes, and only sets `sawn_off` to true.   This is helpful if you want to allow a weapon to be sawed off, but don't want your icon forcibly set to `lefthand_file = 'icons/mob/inhands/weapons/guns_lefthand.dmi` or the weight class set to normal.

## Why It's Good For The Game

Documentation is always good I hear.  Also if you're trying to make a new gun or fix a sprite issue, now you can look at the base type to understand how it works!

## Changelog
:cl:
fix: Mosin Nagant bolt sprites are now visible.  It's like 5 pixels so not a surprise nobody noticed for years.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/ballistic.dm [hunk 1]
```diff
@@ -59,7 +59,14 @@
...
 	var/special_mags = FALSE
-	///The bolt type of the gun, affects quite a bit of functionality, see combat.dm defines for bolt types: BOLT_TYPE_STANDARD; BOLT_TYPE_LOCKING; BOLT_TYPE_OPEN; BOLT_TYPE_NO_BOLT
+	/*
+ 	* The bolt type controls how the gun functions, and what iconstates you'll need to represent those functions.
+ 	* BOLT_TYPE_STANDARD - The Slide doesn't lock back.  Clicking on it will only cycle the bolt.  Only 1 sprite.
+	* BOLT_TYPE_OPEN - Same as standard, but it fires directly from the magazine - No need to rack.  Doesn't hold the bullet when you drop the mag.
+ 	* BOLT_TYPE_LOCKING - This is most handguns and bolt action rifles.  The bolt will lock back when it's empty.  You need yourgun_bolt and yourgun_bolt_locked icon states.
+	* BOLT_TYPE_NO_BOLT - This is shotguns and revolvers.  clicking will dump out all the bullets in the gun, spent or not.
+	* see combat.dm defines for bolt types: BOLT_TYPE_STANDARD; BOLT_TYPE_LOCKING; BOLT_TYPE_OPEN; BOLT_TYPE_NO_BOLT
+ 	*/
```
Comment (Time-Green) [inline code/modules/projectiles/guns/ballistic.dm:69]: [Autodoc this ](https://github.com/tgstation/tgstation/blob/master/.github/guides/AUTODOC.md#documenting-a-proc)

File: code/modules/projectiles/guns/ballistic.dm [hunk 2]
```diff
@@ -92,7 +99,7 @@
 	/// Check if you are able to see if a weapon has a bullet loaded in or not.
 	var/hidden_chambered = FALSE
 
-	///Gun internal magazine modification and misfiring
+	/// Gun internal magazine modification and misfiring ///
```
Comment (Time-Green) [inline code/modules/projectiles/guns/ballistic.dm:102]: This is describing a bunch of variables, so just normal comment this with //

**Review by Time-Green — CHANGES_REQUESTED** [review]
Extremely small nitpicks. Nice work on this!

**Comment by Time-Green** [comment]
Oh right and the linters are bitching about use of spaces for indentation instead of tabs in your new file, can you fix that also?

File: code/modules/projectiles/guns/ballistic.dm (hunk 1 above)
Comment (Stonetear) [reply code/modules/projectiles/guns/ballistic.dm]: fixed.

File: code/modules/projectiles/guns/ballistic.dm (hunk 2 above)
Comment (Stonetear) [reply code/modules/projectiles/guns/ballistic.dm]: Got it.

**Comment by Stonetear** [comment]
@Time-Green Sorry it took me so long to do both of those.  Just pushed both of those changes, plus some other weird fixes I found. 

- Fixes those formatting errors, plus another one I found.
- Fixes a grammatical error in the documentation file.
- De-souls the `boolets` variable name to `bullets` in the ammo counter proc

My new `bolt_types_explained` doesn't have any indentations in it.

**Comment by Stonetear** [comment]
Okay, so it actually gets worse.  For some reason, all `BOLT_TYPE_LOCKING` guns with internal magazines now with an additional bullet, one more than they can hold.  Don't merge this yet until I untangle this.

**Comment by Stonetear** [comment]
Also, I added a parameter for the sawing off proc that removes the hardcoded reference to the default guns inhands image file plus all the other stuff that's snowflaked for the mosin.  By default, it's `handle_modifications = TRUE`, so it changes nothing.  If you wanted to handle all of the stat changes yourself, having this set to false will still handle all the sawing logic for you, but change nothing but `sawn_off = TRUE`.
`//The file might not have a "gun" icon, let's prepare for this`
`lefthand_file = 'icons/mob/inhands/weapons/guns_lefthand.dmi'`
`righthand_file = 'icons/mob/inhands/weapons/guns_righthand.dmi'`
dang, you could lose your mind trying to figure out why it's overwriting your inhands sprite if you wanted to, hypothetically, use a dmi file besides `guns_lefthand.dmi` haha.

**Comment by Time-Green** [comment]
Linters are failing, you need to use tab indentation

Checking for whitespace issues...
 01- space indentation
code/modules/projectiles/guns/ballistic.dm: 	* The bolt type controls how the gun functions, and what iconstates you'll need to represent those functions.
code/modules/projectiles/guns/ballistic.dm: 	* BOLT_TYPE_STANDARD - The Slide doesn't lock back.  Clicking on it will only cycle the bolt.  Only 1 sprite.
code/modules/projectiles/guns/ballistic.dm: 	* BOLT_TYPE_LOCKING - This is most handguns and bolt action rifles.  The bolt will lock back when it's empty.  You need yourgun_bolt and yourgun_bolt_locked icon states.
code/modules/projectiles/guns/ballistic.dm: 	**/

ERROR: Space indentation detected, please use tab indentation.

**Comment by Stonetear** [comment]
Found them, thank you!

**Pull request merged**